Getting started page for usability study (#236)
PowerShell/ProjectMercury: docs/GettingStarted.md

modified the system prompt to say it should act like a PowerShell expert, it is not trained on any
specific PowerShell code or documentation. In the final version you will have to configure the agent
with your endpoint/keys/system prompt before utilizing it. For ease of use we have removed these
steps. 

The Azure agent is an agent designed to bring the Copilot in Azure directly to your CLI and help
provide Azure CLI and Azure PowerShell commands. Typically you will need to login to Azure by using
the `az login` or `Connect-AzAccount` commands but for the sake of this study we have done so
already. Here are some sample queries you can ask to each agent:

### Azure OpenAI Agent

- "How do I create a text file named helloworld in PowerShell?"
- "What is the difference between a switch and a parameter in PowerShell?"
- How do I get the top 10 most CPU intensive processes on my computer?

### Azure Agent
- "How do I create a new resource group with Azure CLI?"
- "How can I list out the storage accounts I have in Azure PowerShell"
- "What is Application Insights?"
- "How to create a web app with Azure CLI?"

Here is a quick demo showing the Azure Agent in action:

> ![A Gif showing Azure Agent in action](/docs/media/AzureAgent.gif)

Reviewer: ```suggestion
![An animation showing Azure Agent in action.](/docs/media/AzureAgent.gif)
```

You can also use a chat command to switch agents. For example, to switch to the `openai-gpt` agent, use `/agent use openai-gpt`.

### Chat commands

By default, `aish` provides a base set of chat `/` commands used to interact with the responses from
the AI model. To get a list of commands, use the `/help` command in the chat session.

```
  Name       Description                                      Source
──────────────────────────────────────────────────────────────────────
  /agent     Command for agent management.                    Core
  /cls       Clear the screen.                                Core
  /code      Command to interact with the code generated.     Core
  /dislike   Dislike the last response and send feedback.     Core
  /exit      Exit the interactive session.                    Core
  /help      Show all available commands.                     Core
  /like      Like the last response and send feedback.        Core
  /refresh   Refresh the chat session.                        Core
  /render    Render a markdown file, for diagnosis purpose.   Core
  /retry     Regenerate a new response for the last query.    Core
```

### Inserting code

> When you are chatting with the agent, you can use the `/code post` command to automatically insert
> the code from the response into the working shell. This is a great way to quickly get the code you
> need to run in your shell. Additionally, you can use the hot key `Ctrl+d, Ctrl+d` to insert the code
> into the working shell.
> 
> ![A Gif showing Inserting Code with AIShell](/docs/media/InsertCode.gif)

Reviewer: ```suggestion
When you are chatting with the agent, you can use the `/code post` command to automatically insert
the code from the response into the working shell. This is the simplest way to quickly get the code
you need to run in your shell. You can also use the hot key <kbd>Ctrl</kbd>+<kbd>d</kbd>,
<kbd>Ctrl</kbd>+<kbd>d</kbd> to insert the code into the working shell.

![An animation showing Inserting Code with AIShell.](/docs/media/InsertCode.gif)

```

### Key bindings for commands

> AIShell supports key bindings for the `/code` command. They are currently hard-coded, but custom key
> bindings will be supported in future releases.
> 
> | Key bindings              | Command          | Functionality |
> | ------------------------- | ---------------- | ------------- |
> | <kbd>Ctrl+d, Ctrl+c</kbd> | `/code copy`     | Copy _all_ the generated code snippets to clipboard |
> | <kbd>Ctrl+\<n\></kbd>     | `/code copy <n>` | Copy the _n-th_ generated code snippet to clipboard |
> | <kbd>Ctrl+d, Ctrl+d</kbd> | `/code post`     | Post _all_ the generated code snippets to the connected application |
> | <kbd>Ctrl+d, \<n\></kbd>  | `/code post <n>` | Post the _n-th_ generated code snippet to the connected application |

Reviewer: ```suggestion
AIShell has key bindings for the `/code` command. They key bindings are currently hard-coded, but
custom key bindings will be supported in future releases.

|                       Key bindings                       |     Command      |                            Functionality                            |
| -------------------------------------------------------- | ---------------- | ------------------------------------------------------------------- |
| <kbd>Ctrl+d</kbd><kbd>Ctrl</kbd>+<kbd>c</kbd>            | `/code copy`     | Copy _all_ the generated code snippets to clipboard                 |
| <kbd>Ctrl</kbd>+<kbd>\<n\></kbd>                         | `/code copy <n>` | Copy the _n-th_ generated code snippet to clipboard                 |
| <kbd>Ctrl</kbd>+<kbd>d</kbd><kbd>Ctrl</kbd>+<kbd>d</kbd> | `/code post`     | Post _all_ the generated code snippets to the connected application |
| <kbd>Ctrl</kbd>+<kbd>d</kbd><kbd>\<n\></kbd>             | `/code post <n>` | Post the _n-th_ generated code snippet to the connected application |
```

### Resolving Errors

> If you encounter an error in your working terminal, you can use the `Resolve-Error` cmdlet to send
> that error to the open AIShell window for resolution. This will allow you to get a response from the
> AI model open to help you resolve the error.

Reviewer: ```suggestion
If you encounter an error in your working terminal, you can use the `Resolve-Error` cmdlet to send
that error to the open AIShell window for resolution. This command asks the AI model to help you
resolve the error.
```

to say it should act like a PowerShell expert. It's not trained on any specific PowerShell code or
documentation. In the final version you have to configure the agent with your endpoint, API keys,
and system prompt before using it. For this walkthrough we've done these steps for you.

The Azure agent is designed brings the Copilot in Azure experience directly to your command line. It
provides assistance for Azure CLI and Azure PowerShell commands. Typically, you need to sign into
Azure using the `az login` or `Connect-AzAccount` commands. However, for this walkthrough we've done
that for you.

## Use AIShell to interact with the agents

Use these sample queries with each agent.

Azure OpenAI Agent

- "How do I create a text file named helloworld in PowerShell?"
- "What is the difference between a switch and a parameter in PowerShell?"
- How do I get the top 10 most CPU intensive processes on my computer?

Azure Agent

- "How do I create a new resource group with Azure CLI?"
- "How can I list out the storage accounts I have in Azure PowerShell"
- "What is Application Insights?"
- "How to create a web app with Azure CLI?"

> Here is a quick demo showing the Azure Agent in action:

Reviewer 2: ```suggestion
Here's a quick demo showing the Azure Agent in action:
```
